ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                    {
+                        continue;
+                    }
+
+                    // Tickers with dots in them like BRK.A and BRK.B appear as BRK-A and BRK-B in Benzinga data.
+                    var symbolTicker = ticker["#text"].ToString().Trim().Replace('-', '.');
+                    var mappedSymbol = _mapFileResolver.ResolveMapFile(symbolTicker, instance.PublicationDate).GetMappedSymbol(instance.PublicationDate);
+
+                    if (string.IsNullOrWhiteSpace(mappedSymbol))
+                    {
+                        Log.Error($"BenzingaDataConverter.Process(): Failed to map old ticker {symbolTicker}. New ticker is null");
+                        continue;
+                    }
+
+                    instance.Symbols.Add(new BenzingaSymbolData()
+                    {
+                        Exchange = exchange,
+                        Sentiment = sentiment,
+                        Symbol = new Symbol(
+                            SecurityIdentifier.GenerateEquity(symbolTicker, Market.USA, mapSymbol: true, mapFileProvider: _mapFileProvider, mappingResolveDate: instance.PublicationDate),
+                            mappedSymbol
+                        )
+                    });
+                }
+
+                yield return instance;
+            }
+        }
+
+        /// <summary>
+        /// Writes the processed data into their respective files
+        /// </summary>
+        /// <param name="news">Enumerable containing news articles</param>
+        /// <param name="date">Date to convert for</param>
+        private void WriteToFile(IEnumerable<BenzingaNews> news, DateTime date)
+        {
+            var finalDirectory = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyMMdd}"));
+            finalDirectory.Create();
+
+            foreach (var article in news)
+            {
+                var createdReference = false;
+
+                // For each ticker, add a reference to the article we're going to store under the "content" folder
+                foreach (var ticker in article.Symbols)
+                {
+                    // Invalid character in path
+                    if (ticker.Symbol.Value.Contains(":"))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} contains invalid character ':'. Skipping");
+                        continue;
+                    }
+                    // ETFs have a special exchange tag associated with them.
+                    // Sometimes, we have an empty exchange for some tickers. Let's try and map them first before giving up on them.
+                    // Example: SPOT (Spotify) on 2018-10-01 has a missing exchange even though it has already IPO'd
+                    if (ticker.Exchange != "NASDAQ" && ticker.Exchange != "NYSE" && ticker.Exchange != "ETF" && !string.IsNullOrEmpty(ticker.Exchange))
```

Comment:
recommend defining something like:

``` csharp
private static readonly IReadOnlyCollection<string> ValidExchanges = new HashSet<string>
{
    "NASDAQ", "NYSE", "ETF"
}
```

do we need this not empty predicate? it kinda threw me... it _looks_ like you're saying the exchange must be one of these three valid exchanges (`NASDAQ`/`NYSE`/`ETF`) -- which means who cares if it's null or empty -- it's not one of the pre-ordained exchanges, so we toss it anyway.

Reply:
The not empty predicate is there to attempt a recovery of the ticker in question if the exchange tag was for some reason omitted. 

`// Example: SPOT (Spotify) on 2018-10-01 has a missing exchange even though it has already IPO'd`

It's a little ugly, so I'll clarify what the purpose of the `string.IsNullOrEmpty` by creating a new parameter for this called `attemptRecovery` 👍